File: dsl-api/src/main/java/cbs/dsl/api/context/HelperContext.java
package cbs.dsl.api.context;

import lombok.Builder;

import java.util.Map;

@Builder(toBuilder = true)
public record HelperContext(
    String eventNumber,
    String performedBy,
    Map<String, Object> params,
    HelperEvaluator helperEvaluator) {

  public HelperContext put(String key, Object value) {
    params.put(key, value);
    return this;
  }

  public Object get(String key) {
    return params.get(key);
  }

  public Object helper(String key, Map<String, Object> values) {
    return helperEvaluator.evaluate(key, values);
  }

  public HelperContext copy() {
    return toBuilder().build();
  }
}

File: dsl-api/src/main/kotlin/cbs/dsl/api/context/HelperContext.kt
package cbs.dsl.api.context

open class HelperContext(
  eventCode: String,
  workflowExecutionId: Long,
  performedBy: String,
  dslVersion: String,
  val params: Map<String, Any>,
) : BaseContext(eventCode, workflowExecutionId, performedBy, dslVersion) {
  open fun helper(name: String, params: Map<String, Any>): Any = Unit

  open fun <T : Any> resolve(clazz: kotlin.reflect.KClass<T>): T =
    error("HelperContext.resolve not implemented")
}